lib/PreferenceList.cc

```diff
@@ -40,7 +40,22 @@ PreferenceList::ConstIterator PreferenceList::find(VertexPtr v) const {
 }
 
 PreferenceList::SizeType PreferenceList::find_index(VertexPtr v) const {
-    return (SizeType) std::distance(cbegin(), find(v));
+    for (auto i = cbegin(); i != cend(); ++i) {
+        if (i->vertex != nullptr) {
```

Comment:
You can change this check to i->vertex == v, splitting this check against a nullptr and v is not needed.

Reply:
If I do that, the current test cases that are using the find_index method are failing.

Reply:
If you can find out the reason of this failure, that'll be great. Is it running into a segfault? Is the argument to find_index ever a nullptr which returns an incorrect value?

Reply:
I think the check is needed because (i->vertex == nullptr) checks whether that particular rank is tied. I had inserted nullptrs inside the preference list at tied ranks, while the tied vertices were inserted inside the tie_map. 
So, for the current test cases, if the vertex does not match, i.e. if (i->vertex != v), it assumes that the rank is tied and goes on to check for the key inside the tie map, which might not exist, thus throwing an error.
More specifically, it calls the get_ties method that uses the .at(key) function when the key is not present.

Reply:
Got it. Can you add this comment to code as well. This'll help when someone is reading the code and it is not obvious why the code was written this way.

Reply:
Sure, done.